pyInterface/package/_likelihood.py

```diff
@@ -43,23 +47,48 @@ def initLikelihood(waveDescThres,
 		pyRootPwa.utils.printErr("could not add acceptance integral. Aborting...")
 		return None
 	accIntFile.Close()
-
+	eventMetas = []
+	openFiles = []
+	for evtFileName in eventFileNameList:
+		evtFile = ROOT.TFile.Open(evtFileName, "READ")
+		if not evtFile:
+			pyRootPwa.utils.printErr("could not open amplitude file '" + evtFileName + "'.")
+			return None
+		evtMeta = pyRootPwa.core.eventMetadata.readEventFile(evtFile)
+		if not evtMeta:
+			pyRootPwa.utils.printErr("could not get metadata for event file '" + evtFileName + "'.")
+			return None
+		eventMetas.append(evtMeta)
+		openFiles.append(evtFile)
+	likelihood.setOnTheFlyBinning(binningMap, eventMetas)
 	for wave in waveDescThres:
+		openFilesWave = []
 		waveName = wave[0]
-		ampFileName = ampFileList[waveName]
-		ampFile = ROOT.TFile.Open(ampFileName, "READ")
-		if not ampFile:
-			pyRootPwa.utils.printErr("could not open amplitude file '" + ampFileName + "'.")
+		ampFileNameList = ampFileList[waveName]
+		if len(eventMetas) > 0 and not len(ampFileNameList) == len(eventMetas):
+			pyRootPwa.utils.printErr("number of amplitude and event files do not match")
 			return None
-		meta = pyRootPwa.core.amplitudeMetadata.readAmplitudeFile(ampFile, waveName)
-		if not meta:
-			pyRootPwa.utils.printErr("could not get metadata for waveName '" + waveName + "'.")
-			return None
-		if (not likelihood.addAmplitude(meta)):
```

Comment:
Remove all changes in .py files, instead change this line to
`if (not likelihood.addAmplitude( [meta] )):`